arm and hppa do not generate pie scripts, so lets fix that.

http://sourceware.org/ml/binutils/2005-03/msg00373.html

--- ld/emulparams/armelf_linux.sh
+++ ld/emulparams/armelf_linux.sh
@@ -8,6 +8,7 @@
 TEMPLATE_NAME=elf32
 EXTRA_EM_FILE=armelf
 GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
 
 DATA_START_SYMBOLS='__data_start = . ;';
 OTHER_TEXT_SECTIONS='*(.glue_7t) *(.glue_7)'
--- ld/emulparams/hppa64linux.sh
+++ ld/emulparams/hppa64linux.sh
@@ -13,6 +13,7 @@
 ENTRY="main"
 TEMPLATE_NAME=elf32
 GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
 
 # We really want multiple .stub sections, one for each input .text section,
 # but for now this is good enough.
--- ld/emulparams/hppalinux.sh
+++ ld/emulparams/hppalinux.sh
@@ -17,5 +17,6 @@
 DATA_PLT=
 PLT_BEFORE_GOT=
 GENERATE_SHLIB_SCRIPT=yes
+GENERATE_PIE_SCRIPT=yes
 TEMPLATE_NAME=elf32
 EXTRA_EM_FILE=hppaelf
